### PR TITLE
Use updateDoc instead of setDoc

### LINKED ISSUE #271: Use updateDoc instead of setDoc
in order to avoid users losing work, please implement : { updateDoc() } catch (e) { if (e.message === “ref doesn’t exist”) setDoc() else throw (e)}

wherever setDoc is currently used.

### PR DESCRIPTION
**Pull request by bhavinsen**
Fixed #271

### CODE REVIEW

File: src/store/item.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const useItemStore = defineStore("item", {
       const item: Item = { ...params, id }
 
       this.itemList.push(item)
-      return this.saveData(podcastname, docname)
+      this.saveData(podcastname, docname)
```
Comment (chrisns) [inline src/store/item.ts:32]:
```suggestion
      return this.saveData(podcastname, docname)
```

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — APPROVED** [review]

File: src/store/item.ts [hunk 2]
```diff
@@ -51,10 +53,20 @@ export const useItemStore = defineStore("item", {
 
     async saveData(podcastname: string, docname: string) {
       const docRef = doc(collection(db, podcastname), docname)
-      return setDoc(docRef, {
-        items: this.itemList,
-        slotTitles: this.slotTitleList,
-      })
+      try {
+        await updateDoc(docRef, {
+          items: this.itemList,
```
Comment (chrisns) [inline src/store/item.ts:58]: The real point of this exercise is to use arrayUnion etc so you're only pushing the change not the whole doc each time

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/store/item.ts (hunk 2 above)
Comment (bhavinsen) [reply src/store/item.ts:58]: I have tried arrayUnion but it duplicates the record.
would you please check the attached video?

https://user-images.githubusercontent.com/13918292/171849055-2efdd3a5-81ad-48ae-8045-8e6b7fbe2d43.mp4

Comment (bhavinsen) [reply src/store/item.ts:79]: I have used arrayUnion/arrayRemove in creating/deleting an item 

Also, I have tried using arrayUnion in the update/drag item function but is not working well so I have not used arrayUnion in the update/drag item.

File: cypress/e2e/podcasts.cy.js [hunk 3]
```diff
@@ -183,8 +183,15 @@ describe("newsdesk logged in", () => {
     cy.get("section[slotno=7] ul li div a")
       .invoke("attr", "href")
       .should("eq", link)
-    cy.get("section[slotno=7] ul li div a").click()
-    cy.get("section[slotno=7] button[title='Delete']").click()
+    cy.get("section[slotno=7]").should(
+      "contain",
+      "https://twitter.com/PoliticusSarah/status/15207595...",
+    )
+    window.open(link)
```
Comment (chrisns) [inline cypress/e2e/podcasts.cy.js:190]: whats this doing?

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — CHANGES_REQUESTED** [review]
not sure what the window.open is doing in the cypress tests?

**Pull request merged**